server/src/test/java/org/elasticsearch/gateway/GatewayMetaStatePersistedStateTests.java

```diff
@@ -312,7 +312,7 @@ public void testStatePersistedOnLoad() throws IOException {
         for (Path path : paths) {
             Settings settings = Settings.builder()
                 .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toAbsolutePath())
-                .put(Environment.PATH_DATA_SETTING.getKey(), path.toString()).build();
+                .put(Environment.PATH_DATA_SETTING.getKey(), path.getParent().getParent().toString()).build();
```

Comment:
This test used to fail, fixed diff in backport
https://github.com/crate/crate/pull/11270/files#diff-d877765155c7bdc98f0d08ca7ad9e9c8290cbfe8cdb097db4937ca52fbb112b4R314
and 
https://github.com/elastic/elasticsearch/pull/50928/files#diff-d877765155c7bdc98f0d08ca7ad9e9c8290cbfe8cdb097db4937ca52fbb112b4R313